magenta/models/polyphonic_rnn/polyphonic_rnn_graph.py

```diff
@@ -0,0 +1,179 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Manages the model's graph."""
+
+# internal imports
+
+import numpy as np
+import tensorflow as tf
+
+from magenta.models.polyphonic_rnn import polyphonic_rnn_lib
+
+
+class Graph(object):
+
+  def __init__(self, examples):
+    self.num_epochs = 500
+    self.batch_size = 32
+    # ~1 step per seconds
+    # 30 steps ~= 30 seconds
+    sequence_length = 30
+
+    self.train_itr = polyphonic_rnn_lib.TFRecordDurationAndPitchIterator(
+        examples, self.batch_size, stop_index=.9,
+        sequence_length=sequence_length)
+
+    duration_mb, note_mb = next(self.train_itr)
+    self.train_itr.reset()
+
+    self.valid_itr = polyphonic_rnn_lib.TFRecordDurationAndPitchIterator(
```

Comment:
Do the other RNN models hold out part of user data as a validation set? Or should we just train on the whole thing? In the extreme could even early stop, then retrain with all data to the same number of epochs. I don't know what the "normal" thing is for Magenta models